frontend/src/components/dashboard/dashboard.tsx

```diff
@@ -33,15 +33,36 @@ const Dashboard: FC = () => {
     dispatch(dashboardActions.getCategories());
   }, [dispatch]);
 
-  if (dataStatus === DataStatus.PENDING) {
-    return <Spinner />;
-  }
-
   const handleNewCourseModalToggle = (evt: React.MouseEvent | void): void => {
     evt?.stopPropagation();
     setIsNewCourseModalOpen(!isNewCourseModalOpen);
   };
 
+  if (dataStatus === DataStatus.PENDING) {
+    return (
+      <div className={styles.dashboard}>
+        <div className={styles.headerWrapper}>
+          <div className={styles.header}>
+            <h1 className={styles.headingText}>Courses</h1>
+            {hasUser && (
+              <Button
+                label="+ Add new course"
+                btnColor="blue"
+                onClick={handleNewCourseModalToggle}
+              />
+            )}
+            <AddCourseModal
+              isModalOpen={isNewCourseModalOpen}
+              onModalToggle={handleNewCourseModalToggle}
+            />
+          </div>
+          <CategoriesList items={categories} />
+        </div>
+        <Spinner />
+      </div>
+    );
+  }
```

Comment:
Using a ternary on one line would prevent you of duplicating other ones
```suggestion
    return (
      <div className={styles.dashboard}>
        <div className={styles.headerWrapper}>
          <div className={styles.header}>
            <h1 className={styles.headingText}>Courses</h1>
            {hasUser && (
              <Button
                label="+ Add new course"
                btnColor="blue"
                onClick={handleNewCourseModalToggle}
              />
            )}
            <AddCourseModal
              isModalOpen={isNewCourseModalOpen}
              onModalToggle={handleNewCourseModalToggle}
            />
          </div>
          <CategoriesList items={categories} />
        </div>
        {dataStatus === DataStatus.PENDING ? <Spinner /> : <CoursesList courses={courses} />}
      </div>
    );
```

Reply:
agreed